Added changes to terraform integration program (#156)
hashicorp/terraform-docs-common: website/docs/docs/partnerships.mdx

> The Terraform Integration Program facilitates prospect partners in creating and publishing Terraform integrations that have been verified by HashiCorp.

Reviewer: ```suggestion
The Terraform Integration Program facilitates prospect partners in creating and publishing Terraform integrations verified by HashiCorp.
```

Suggest changing "that have been verified" to "verified' to avoid a passive voice style issue.

Author: I'll change the verbiage to validated actually

## Types of Terraform Integrations

The Terraform ecosystem is designed to enable users to apply Terraform across different use cases and environments. The Terraform Integration Program currently supports both workflow and integration partners (details below). Some partners can be both, depending on their use cases.

> - **Workflow Partners** build integrations for Terraform Cloud and/or Terraform Enterprise. Typically, these partners are seeking to enable customers to use their existing platform within a Terraform Run.

Reviewer: ```suggestion
- **Workflow Partners** build integrations for Terraform Cloud, Terraform Enterprise, or both. Typically, these partners seek to enable customers to use their existing platform within a Terraform Run.
```

Suggest avoiding "and/or" and suggest changing "are seeking" to "seek" to use fewer words.

- No code/low code: Partners focused on implementing, deployment and delivery of IT, Supply Chain, Operations Management, Business Management and other workflows
- **SSO (Single Sign On):** Partners focused on authentication for end users to securely sign on.
- **CI/CD:** Partners focused on continuous integration and continuous delivery/deployment.
- **VCS:** Partners focused on tracking and managing software code changes.

Most workflow partners create [run tasks](/cloud-docs/integrations/run-tasks) that let them integrate their services directly into the Terraform workflow. [Run tasks](/cloud-docs/integrations/run-tasks) let Terraform Cloud execute tasks in external systems at certain points in the Terraform Cloud run lifecycle. Specifically, Terraform Cloud users can configure run tasks to execute during the post-plan and the pre-plan (public beta) run stages. Refer to the [Terraform Cloud Integrations](/docs/partnerships#terraform-cloud-integrations) or [Terraform Enterprise Integrations](/docs/partnerships#terraform-enterprise-integrations) documentation for more details.

> Most workflow partners create run tasks that let them integrate their services directly into the Terraform workflow. Most workflow partners integrate with the Terraform workflow itself. Run tasks allow Terraform Cloud to execute tasks in external systems at specific points in the Terraform Cloud run lifecycle. This offers much more extensibility to Terraform Cloud customers, enabling them to integrate your services into the Terraform Cloud workflow. This feature allows users to add and execute these tasks during the post-plan and the new pre-plan & pre-apply stages. Eventually, we will open the third phase of this workflow to Terraform Cloud users, the post-apply stage. Reference the [Terraform Cloud Integrations](/docs/partnerships#terraform-cloud-integrations) or [Terraform Enterprise Integrations](/docs/partnerships#terraform-enterprise-integrations) documentation for more details.

Reviewer: ```suggestion
Most workflow partners create run tasks that let them integrate their services directly into the Terraform workflow. Most workflow partners integrate with the Terraform workflow itself. Run tasks allow Terraform Cloud to execute tasks in external systems at specific points in the Terraform Cloud run lifecycle. This integration offers much more extensibility to Terraform Cloud customers, enabling them to integrate your services into the Terraform Cloud workflow. This feature allows users to add and execute these tasks during the post-plan and the new pre-plan and pre-apply stages. Eventually, we will open the third phase of this workflow to Terraform Cloud users, the post-apply stage. Refer to the [Terraform Cloud Integrations](/docs/partnerships#terraform-cloud-integrations) or [Terraform Enterprise Integrations](/docs/partnerships#terraform-enterprise-integrations) documentation for more details.
```
Suggest adding "integration" to avoid a "This" without a subject.

> Vendors who choose not to support their provider and prefer to make it a community-supported provider will no longer be listed as Partner.

Reviewer: ```suggestion
Vendors who choose not to support their provider and prefer to make it a community-supported provider cannot be Partners.
```

Suggest changing "will no longer be listed as Partner" to "cannot be Partners" to avoid future tense and passive voice style issues.

Author: @mgarrell777 -- they can still be technology partners but their provider will be de-listed as "partner" supported tag.

could you help me phrase this better?

> -> Terraform modules published by Partner: We now have the ability to tag partner published modules in the Terraform Registry. If you have a modules that you’d like to receive the “Partner” tag, please reach out to us at technologypartners@hashicorp.com

Reviewer: ```suggestion
-> Terraform modules published by Partner: We now have the ability to tag partner published modules in the Terraform Registry. If you have a module that you want to receive the “Partner” tag, please contact technologypartners@hashicorp.com
```

Suggest changing "you'd like" to "you want" to avoid contractions. Suggest changing "reach out to us" to "contact" to use fewer words.

## Terraform Cloud Integrations

Run tasks allow Terraform Cloud to execute tasks in external systems at certain points in the Terraform Cloud run lifecycle. Specifically, Terraform Cloud users can add and execute these tasks during the pre-plan (public beta), pre-apply (public beta) and post-plan run stages. To execute a run task, Terraform Cloud sends an API payload to the external system. This payload contains a collection of run-related information and a callback URL that the external system can use to send updates back to Terraform Cloud.

The external system can then use this run information and respond back to Terraform Cloud with a passed or failed status. Terraform Cloud uses this status response to determine if a run should proceed, based on the task's enforcement settings within a workspace.

> Partners who successfully complete the Terraform Cloud integration checklist obtain a Terraform Cloud badge. This signifies HashiCorp has validated the integration under ideal conditions and the partner is a member of the HashiCorp Technology Partner Program.

Reviewer: ```suggestion
Partners who successfully complete the Terraform Cloud integration checklist obtain a Terraform Cloud badge. This badge signifies HashiCorp validated the integration under ideal conditions and the partner is a member of the HashiCorp Technology Partner Program.
```

Suggest adding "badge" to avoid a "This" with no subject. Suggest changing "has validated" to "validated" to avoid a passive voice style issue.

> - Note: As of September 2022, pre-plan & pre-apply run tasks are available only as a public beta feature, are subject to change, and not all customers will see this functionality in their Terraform Cloud organization. This feature is enabled by default for our business tier customers of Terraform Cloud. If you have a customer that is interested in run tasks and are not a current Terraform Cloud for Business customer, please ask them to [sign up for the public beta](https://docs.google.com/forms/d/e/1FAIpQLSf3JJIkU05bKWov2wXa9c-QV524WNaHuGIk7xjHnwl5ceGw2A/viewform).

Reviewer: ```suggestion
- Note: As of September 2022, pre-plan and pre-apply run tasks are available only as a public beta feature, are subject to change, and not all customers see this functionality in their Terraform Cloud organization. This feature's enablement is the default for our Terraform Cloud Business tier customers. If you have a customer who wants to run tasks and are not a current Terraform Cloud Business tier customer, please ask them to [sign up for the public beta](https://docs.google.com/forms/d/e/1FAIpQLSf3JJIkU05bKWov2wXa9c-QV524WNaHuGIk7xjHnwl5ceGw2A/viewform).
```
Suggest avoid ampersands and using the word "and" instead. Suggest removing "will" to avoid future tense. Suggest changing "This feature is enabled by default" to "This feature's enablement is the default" to avoid a passive voice style issue. Suggest modifying the tier name to "Terraform Cloud Business tier" for uniformity. Suggest changing "is interested in" to "wants to" to avoid a passive voice style issue.

> ### 2. Develop & Test

Reviewer: ```suggestion
### 2. Develop and Test
```

Suggest avoiding ampersands and using the word "and" instead.

Partners should build an integration using the [run tasks APIs in Terraform Cloud](/cloud-docs/api-docs/run-tasks). To learn more about how post-plan run tasks can enhance the Terraform Cloud workflow, refer to the following diagram and read our [run tasks announcement](https://www.hashicorp.com/blog/terraform-cloud-run-tasks-beta-now-available). For a more hands-on experience with run tasks follow the [HCP Packer run tasks learn guide here](https://learn.hashicorp.com/tutorials/packer/setup-tfc-run-task?in=terraform/cloud).

For the run tasks API documentation [click here](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run-tasks/run-tasks).

![RunTask Diagram](/img/docs/post-plan-runtask-diagram.png)

> -> **Note:** Pre-plan & pre-apply run tasks are only available as a public beta. [Updated September 1, 2022]

Reviewer: ```suggestion
-> **Note:** Pre-plan and pre-apply run tasks are only available as a public beta. [Updated September 1, 2022]
```

Suggest avoiding ampersands and using the word "and" instead.

> To better understand how pre-plan run task enhances the Terraform workflow, see diagram listed below. For additional API resources, [click here](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run-tasks/run-task-stages-and-results).

Reviewer: ```suggestion
To better understand how pre-plan run task enhances the Terraform workflow, refer to the diagram of pre-plan tasks. For additional API resources, [click here](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run-tasks/run-task-stages-and-results).
```

Suggest changing "see" to "refer to" to avoid ableist language. Suggest removing "below" to avoid directional language and to more specifically tell the user which diagram to find.

-> Contact us at [technologypartners@hashicorp.com](mailto:technologypartners@hashicorp.com) with any questions or feedback.

After successfully building a Run Tasks integration with Terraform Cloud, partners should contact their Alliance Manager to validate the integration against Terraform Enterprise. Many customers deploy Terraform Enterprise in a private cloud or air-gapped environment, so partners must ensure that their product can communicate bi-directionally with Terraform Enterprise inside these types of networks.

> Upon successfully building Run Tasks integration with Terraform Cloud partners can reach out to their Alliance Manager to validate it against Terraform Enterprise. It’s common for customers to deploy Terraform Enterprise in a private cloud or in an air-gapped environment in which case partners need ensure that their product can communicate bi-directionally with TFE inside the customer’s network.

Reviewer: ```suggestion
Upon successfully building Run Tasks integration with Terraform Cloud partners can contact their Alliance Manager to validate it against Terraform Enterprise. Customers commonly deploy Terraform Enterprise in a private cloud or in an air-gapped environment, in which case partners need tp ensure that their product can communicate bi-directionally with TFE inside the customer’s network.
```

Suggest changing "reach out to" to "contact" to use fewer words. Suggest changing "It's common for customers to" to "Customers commonly" to use fewer words and avoid contractions.

Author: I think there's a misspelling here tp -> to, I'll take this suggestion in a separate commit
